### ISSUE TITLE
Django 1.8b1: relation "auth_user" does not exist

### ISSUE THREAD

**Issue by ivasic** [evonove/django-oauth-toolkit#204]
Seems that something is bothering Django 1.8b1 when applying migrations on PostgreSQL, even on a fresh django project without any apps (or any fancy stuff with custom models). Just create a new django project, add oauth2_provider to the installed apps and run "python manage.py test" to trigger it.

I've created a simple project that reproduces it here: http://cl.ly/3e1b0B1g0n2J (make sure you create a PostgreSQL db as per settings.py)

I should note that when I remove oauth2_provider from my installed apps, do the first migration (so that the auth_user table is created), put it back in installed apps and do another migration - then it works fine since auth_user table is created beforehand. Running tests, however, creates everything at once so it always triggers this problem and I'm unable to run any tests.

Environment: Django 1.8b1, django-oauth-toolkit 0.7.2, Python 3.4

Anyone knows if this is a Django 1.8 problem perhaps?

**Comment by chrisguitarguy** [comment]
This doesn't happen with Django 1.7.

`manage.py test` runs the `migrate` command, which fails with the same error:

> django.db.utils.ProgrammingError: relation "auth_user" does not exist

From what I can tell, this happening because the `syncdb` Stuff, which creates the models for this library, runs before the migrations. The docs warn against this: https://docs.djangoproject.com/en/1.7/topics/migrations/#dependencies

> Be aware, however, that unmigrated apps cannot depend on migrated apps, by the very nature of not having migrations. This means that it is not generally possible to have an unmigrated app have a ForeignKey or ManyToManyField to a migrated app; some cases may work, but it will eventually fail.

But there's a way around it!

> As time goes on, more and more third-party apps will get migrations, but in the meantime you can either **give them migrations yourself (using MIGRATION_MODULES to store those modules outside of the app’s own module if you wish)**, [...]

So in your settings, you can add something like this:

```python
# makemigrations requires this for some reason or it errors
# Just set to the default value
OAUTH2_PROVIDER_APPLICATION_MODEL = 'oauth2_provider.Application'

# tell django where to put the oauth2 migrations
MIGRATION_MODULES = {
   # key: app name, value: a fully qualified package name, not the usual `app_label.something_else`
  'oauth2_provider': 'yourapp.migrations.oauth2_provider',
}
```

Then run `./manage.py makemigrations oauth2_provider` to generate a new migration for this app.

Long term the solution seems to be migrating to django's internal migration system. I've not used south before, so not sure how much of a pain that is.

**Comment by ivasic** [comment]
Thanks Christopher, this worked great. Explicit migrations for oauth2_provider seem to solve the problem. I appreciate you looking into this one.

And indeed, all seems to work fine with Django 1.7. I'm experiencing this behaviour only on 1.8.

**Comment by dustinfarris** [comment]
Also see #224

**Comment by hicolour** [comment]
The best work around I found was, to run first of wall migration for auth

```
migrate auth 
migrate
```

**Comment by arsenalstriker14** [comment]
hicolours solution was an easy fix for me as well on 1.8.2

**Comment by jleclanche** [comment]
This is a non-issue now.
